### ISSUE TITLE
Proper multi-threading

### ISSUE THREAD

**Issue by aimalz** [aimalz/chippr#70]
My tests currently run in parallel via external scripts, not using an inherent `chippr` mode.  The reason is that the function that evaluates the posterior over n(z) is not pickleable, so `emcee` multi-threading doesn't work.  This issue is for either choosing a sampler that somehow doesn't care about this shortcoming (I suspect there may not be such a thing) or making the function pickleable.  It's a low priority for the scopes of paper 0 (the model and proof of concept) and paper 1 (demonstration on toy test cases) but essential for paper 2 (more realistic and/or real data at scale).

**Comment by aimalz** [comment]
Note: I will check if I can use [`pathos`](http://trac.mystic.cacr.caltech.edu/project/pathos/wiki.html) to get around the issue.
